From d6c4d6d0f74fafcefe1ca2f9a1adfe03968378b2 Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <chenxiaolong@cxl.epac.to>
Date: Sun, 24 Dec 2017 18:31:16 -0500
Subject: [PATCH 2/5] Remove support for arbitrary-length properties

NDK doesn't have an API 26 libc.a yet.

This partially reverts 4eacd70f85ec2ffc071d06945524c40967ce30f7

Signed-off-by: Andrew Gunnerson <chenxiaolong@cxl.epac.to>
---
 libcutils/properties.cpp | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/libcutils/properties.cpp b/libcutils/properties.cpp
index d2645e640..d7bf885a5 100644
--- a/libcutils/properties.cpp
+++ b/libcutils/properties.cpp
@@ -129,13 +129,12 @@ struct callback_data {
     void* cookie;
 };
 
-static void trampoline(void* raw_data, const char* name, const char* value, unsigned /*serial*/) {
-    callback_data* data = reinterpret_cast<callback_data*>(raw_data);
-    data->callback(name, value, data->cookie);
-}
-
 static void property_list_callback(const prop_info* pi, void* data) {
-    __system_property_read_callback(pi, trampoline, data);
+    char name[PROP_VALUE_MAX];
+    char value[PROP_VALUE_MAX];
+    __system_property_read(pi, name, value);
+    callback_data* userdata = static_cast<callback_data *>(data);
+    userdata->callback(name, value, userdata->cookie);
 }
 
 int property_list(void (*fn)(const char* name, const char* value, void* cookie), void* cookie) {
-- 
2.14.3

